fastestimator/trace/adapt/lr_scheduler.py

```diff
@@ -50,25 +51,53 @@ class LRScheduler(Trace):
 
     def __init__(self, model: Union[tf.keras.Model, torch.nn.Module], lr_fn: Callable[[int], float]) -> None:
         self.model = model
-        self.lr_fn = lr_fn
-        assert hasattr(lr_fn, "__call__"), "lr_fn must be a function"
-        arg = list(inspect.signature(lr_fn).parameters.keys())
-        assert len(arg) == 1 and arg[0] in {"step", "epoch"}, "the lr_fn input arg must be either 'step' or 'epoch'"
-        self.schedule_mode = arg[0]
-        super().__init__(mode="train", outputs=self.model.model_name + "_lr")
+        self.lr_fn = ARC() if lr_fn == "arc" else lr_fn
```

Comment:
given that arc behaves completely differently than a regular lr_fn, we may be better off putting all of it into a single new trace rather than trying to make a hybrid trace/scheduler approach. especially since the arc api has to be invoked in such a specific order. If we definitely want it this way maybe we can come up with a more generalizable api to interact with complex schedulers?

Reply:
ARC is a LR schedule, so it should be used in LR scheduler trace. Otherwise we will need to worry about what happens when user use both ARC and LR scheduler trace. 

As for the generalizable API, we will worry about it when we have other scheduler similar to ARC, then we will know the requirements of designing that API.